File: C++SrcFiles/Reaper_shoot.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
#include <algorithm>

using namespace std;

class ReaperShoot : public MonoBehaviour
{
private:
    GameObject* attackPrefab;
    Transform* attackPosition;
    float attackTimer;

public:
    ReaperShoot()
    {
        attackPrefab = Resources::Load<GameObject>("Prefabs/ReaperAttack");
        attackPosition = transform->Find("AttackPosition");
        attackTimer = 0.0f;
    }

    void Update()
    {
        // Count up the attack timer
        attackTimer += Time.deltaTime;

        // Check if it's time to attack
        if (attackTimer >= 3.0f) {
            attackTimer = 0.0f;
            Shoot();
        }
    }

    void Shoot()
    {
        // Instantiate an attack object at the attack position
        Instantiate(attackPrefab, attackPosition->position, Quaternion::identity);
    }
};

File: C#SrcFiles/Reaper_shoot.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Reaper_shoot : MonoBehaviour
{
    // the prefab of the reaperattack png;
    public GameObject attack;
    //position of the attack;
    public Transform attackposition;

    // Intervals between attacks;
    private float attacktimer;
    void Update()
    {
        // counting till 3 seconds and then shooting the reaper bullet.
        attacktimer += Time.deltaTime;
        if (attacktimer > 3)
        {
            attacktimer = 0;
            shoot();
        }
    }


    private void shoot()
    {
        Instantiate(attack, attackposition.position, Quaternion.identity);
    }



}
